hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AFileSystemIsolatedClassloader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a;
+
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestS3AFileSystemIsolatedClassloader {
+
+  public static class CustomClassLoader extends ClassLoader {
+  }
+
+  private final ClassLoader classLoader = new CustomClassLoader();
+
+  @Test
+  public void isolatedClasspath() throws IOException {
+    try (S3AFileSystem fs = new S3AFileSystem()) {
+      Configuration conf = new Configuration();
```

Comment:
I actually could not use getConfiguration() otherwise Configuration object would've been initialised in a way that made my tests fail, I hope the way I did it is still good enough